diff --git a/node_modules/@firebase/auth-compat/dist/index.esm2017.js b/node_modules/@firebase/auth-compat/dist/index.esm2017.js
index 51b8a80..feae2e4 100644
--- a/node_modules/@firebase/auth-compat/dist/index.esm2017.js
+++ b/node_modules/@firebase/auth-compat/dist/index.esm2017.js
@@ -877,11 +877,11 @@ function buildPersistenceHierarchy(apiKey, appName) {
  * limitations under the License.
  */
 class PhoneAuthProvider {
-    constructor() {
+    constructor(auth = firebase.auth()) {
         this.providerId = 'phone';
         // TODO: remove ts-ignore when moving types from auth-types to auth-compat
         // @ts-ignore
-        this._delegate = new exp.PhoneAuthProvider(unwrap(firebase.auth()));
+        this._delegate = new exp.PhoneAuthProvider(unwrap(auth));
     }
     static credential(verificationId, verificationCode) {
         return exp.PhoneAuthProvider.credential(verificationId, verificationCode);
