diff --git a/dist/index.js b/dist/index.js
index b31914a9af7bc1d5dd8b3dc7eae5ba5ba71638fe..af301710aaad38fda6187b9fe3a722dddb99d04e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -303,7 +303,7 @@ function isBrowser() {
 function createBrowserSupabaseClient({
   supabaseUrl,
   supabaseKey,
-  options,
+  options: {auth, ...options} = {}, // @see https://github.com/supabase/auth-helpers/pull/449,
   cookieOptions: {
     name = "supabase-auth-token",
     domain,
@@ -316,6 +316,7 @@ function createBrowserSupabaseClient({
   return (0, import_supabase_js.createClient)(supabaseUrl, supabaseKey, {
     ...options,
     auth: {
+      ...auth,
       storageKey: name,
       storage: {
         getItem(key) {