diff --git a/dist/index.js b/dist/index.js
index eedc39c..f05f046 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -39,8 +39,8 @@ const pt = __importStar(require("./locales/pt"));
 exports.pt = pt;
 const nl = __importStar(require("./locales/nl"));
 exports.nl = nl;
-const zh = __importStar(require("./locales/zh"));
-exports.zh = zh;
+// const zh = __importStar(require("./locales/zh"));
+// exports.zh = zh;
 exports.strict = en.strict;
 exports.casual = en.casual;
 function parse(text, ref, option) {
