diff --git a/node_modules/firebase/compat/index.d.ts b/node_modules/firebase/compat/index.d.ts
index d28e37f..8808e36 100644
--- a/node_modules/firebase/compat/index.d.ts
+++ b/node_modules/firebase/compat/index.d.ts
@@ -8219,7 +8219,9 @@ declare namespace firebase.firestore {
    * paths (e.g. 'foo' or 'foo.baz') mapped to values. Fields that contain dots
    * reference nested fields within the document.
    */
-  export type UpdateData = { [fieldPath: string]: any };
+  export type UpdateData<T, U> = Exclude<keyof U, keyof T> extends never
+    ? {[K in keyof U]?: U[K] | FieldValue}
+    : {[K in keyof T]?: T[K] | FieldValue};
 
   /**
    * Constant used to indicate the LRU garbage collection should be disabled.
@@ -8471,7 +8473,7 @@ declare namespace firebase.firestore {
      * @param collectionPath A slash-separated path to a collection.
      * @return The `CollectionReference` instance.
      */
-    collection(collectionPath: string): CollectionReference<DocumentData>;
+    collection<T>(collectionPath: string): CollectionReference<T>;
 
     /**
      * Gets a `DocumentReference` instance that refers to the document at the
@@ -8480,7 +8482,7 @@ declare namespace firebase.firestore {
      * @param documentPath A slash-separated path to a document.
      * @return The `DocumentReference` instance.
      */
-    doc(documentPath: string): DocumentReference<DocumentData>;
+    doc<T>(documentPath: string): DocumentReference<T>;
 
     /**
      * Creates and returns a new Query that includes all documents in the
@@ -8492,7 +8494,7 @@ declare namespace firebase.firestore {
      * will be included. Cannot contain a slash.
      * @return The created Query.
      */
-    collectionGroup(collectionId: string): Query<DocumentData>;
+    collectionGroup<T>(collectionId: string): Query<T>;
 
     /**
      * Executes the given `updateFunction` and then attempts to commit the changes
@@ -8959,7 +8961,7 @@ declare namespace firebase.firestore {
      */
     set<T>(
       documentRef: DocumentReference<T>,
-      data: Partial<T>,
+      data: import('@alka/util').ObjectPartialDeep<T>,
       options: SetOptions
     ): Transaction;
 
@@ -8985,7 +8987,7 @@ declare namespace firebase.firestore {
      * within the document.
      * @return This `Transaction` instance. Used for chaining method calls.
      */
-    update(documentRef: DocumentReference<any>, data: UpdateData): Transaction;
+    update<T, U>(documentRef: DocumentReference<T>, data: UpdateData<T, U>): Transaction;
 
     /**
      * Updates fields in the document referred to by the provided
@@ -9044,7 +9046,7 @@ declare namespace firebase.firestore {
      */
     set<T>(
       documentRef: DocumentReference<T>,
-      data: Partial<T>,
+      data: import('@alka/util').ObjectPartialDeep<T>,
       options: SetOptions
     ): WriteBatch;
 
@@ -9070,7 +9072,7 @@ declare namespace firebase.firestore {
      * within the document.
      * @return This `WriteBatch` instance. Used for chaining method calls.
      */
-    update(documentRef: DocumentReference<any>, data: UpdateData): WriteBatch;
+    update<T, U>(documentRef: DocumentReference<T>, data: UpdateData<T, U>): WriteBatch;
 
     /**
      * Updates fields in the document referred to by this `DocumentReference`.
@@ -9217,7 +9219,7 @@ declare namespace firebase.firestore {
      * @param collectionPath A slash-separated path to a collection.
      * @return The `CollectionReference` instance.
      */
-    collection(collectionPath: string): CollectionReference<DocumentData>;
+    collection<U>(collectionPath: string): CollectionReference<U>;
 
     /**
      * Returns true if this `DocumentReference` is equal to the provided one.
@@ -9237,7 +9239,7 @@ declare namespace firebase.firestore {
      * @return A Promise resolved once the data has been successfully written
      * to the backend (Note that it won't resolve while you're offline).
      */
-    set(data: Partial<T>, options: SetOptions): Promise<void>;
+    set(data: import('@alka/util').ObjectPartialDeep<T>, options: SetOptions): Promise<void>;
 
     /**
      * Writes to the document referred to by this `DocumentReference`. If the
@@ -9260,7 +9262,7 @@ declare namespace firebase.firestore {
      * @return A Promise resolved once the data has been successfully written
      * to the backend (Note that it won't resolve while you're offline).
      */
-    update(data: UpdateData): Promise<void>;
+    update<U>(data: UpdateData<T, U>): Promise<void>;
 
     /**
      * Updates fields in the document referred to by this `DocumentReference`.
@@ -10014,7 +10016,7 @@ declare namespace firebase.firestore {
      * @param documentPath A slash-separated path to a document.
      * @return The `DocumentReference` instance.
      */
-    doc(documentPath?: string): DocumentReference<T>;
+    doc<U = T>(documentPath?: string): DocumentReference<U>;
 
     /**
      * Add a new document to this collection with the specified data, assigning
