diff --git a/dist/es/pipe.d.ts b/dist/es/pipe.d.ts
index 76751cb628db9a0fe45067a5aafe081346467a99..08764fbbebe23be65551af8a0c2fd3e16bcb8165 100644
--- a/dist/es/pipe.d.ts
+++ b/dist/es/pipe.d.ts
@@ -14,6 +14,7 @@
  * @data_first
  * @category Function
  */
+export declare function pipe<A>(value: A): A;
 export declare function pipe<A, B>(value: A, op1: (input: A) => B): B;
 export declare function pipe<A, B, C>(value: A, op1: (input: A) => B, op2: (input: B) => C): C;
 export declare function pipe<A, B, C, D>(value: A, op1: (input: A) => B, op2: (input: B) => C, op3: (input: C) => D): D;