diff --git a/node_modules/remeda/dist/es/pipe.d.ts b/node_modules/remeda/dist/es/pipe.d.ts
index 76751cb..2780083 100644
--- a/node_modules/remeda/dist/es/pipe.d.ts
+++ b/node_modules/remeda/dist/es/pipe.d.ts
@@ -14,6 +14,7 @@
  * @data_first
  * @category Function
  */
+export declare function pipe<A>(value: A): A;
 export declare function pipe<A, B>(value: A, op1: (input: A) => B): B;
 export declare function pipe<A, B, C>(value: A, op1: (input: A) => B, op2: (input: B) => C): C;
 export declare function pipe<A, B, C, D>(value: A, op1: (input: A) => B, op2: (input: B) => C, op3: (input: C) => D): D;
diff --git a/node_modules/remeda/src/pipe.ts b/node_modules/remeda/src/pipe.ts
index b925000..9dd2df5 100644
--- a/node_modules/remeda/src/pipe.ts
+++ b/node_modules/remeda/src/pipe.ts
@@ -16,6 +16,7 @@ import { LazyResult } from './_reduceLazy';
  * @data_first
  * @category Function
  */
+export function pipe<A>(value: A): A;
 export function pipe<A, B>(value: A, op1: (input: A) => B): B;
 export function pipe<A, B, C>(
   value: A,
