diff --git a/node_modules/typescript/lib/tsserver.js b/node_modules/typescript/lib/tsserver.js
index 2cd2108..b885670 100644
--- a/node_modules/typescript/lib/tsserver.js
+++ b/node_modules/typescript/lib/tsserver.js
@@ -14080,7 +14080,7 @@ var ts;
 (function (ts) {
     ts.resolvingEmptyArray = [];
     ts.externalHelpersModuleNameText = "tslib";
-    ts.defaultMaximumTruncationLength = 160;
+    ts.defaultMaximumTruncationLength = 16000; // patch for https://github.com/microsoft/TypeScript/issues/35601
     ts.noTruncationMaximumTruncationLength = 1000000;
     function getDeclarationOfKind(symbol, kind) {
         var declarations = symbol.declarations;
