diff --git a/lib/tsserver.js b/lib/tsserver.js
index 481cb5866634edf6d8c82fc9f0e272ce5268c4de..f2fa4dbe84349f5d05a56a6655d0f27b9b711786 100644
--- a/lib/tsserver.js
+++ b/lib/tsserver.js
@@ -14335,7 +14335,7 @@ var ts;
 (function (ts) {
     ts.resolvingEmptyArray = [];
     ts.externalHelpersModuleNameText = "tslib";
-    ts.defaultMaximumTruncationLength = 160;
+    ts.defaultMaximumTruncationLength = 16000; // patch for https://github.com/microsoft/TypeScript/issues/35601
     ts.noTruncationMaximumTruncationLength = 1000000;
     function getDeclarationOfKind(symbol, kind) {
         var declarations = symbol.declarations;