diff --git a/lib/types.d.ts b/lib/types.d.ts
index b54c8ed..1ac2ad6 100644
--- a/lib/types.d.ts
+++ b/lib/types.d.ts
@@ -34,6 +34,7 @@ export declare type SafeParseSuccess<Output> = {
 export declare type SafeParseError<Input> = {
     success: false;
     error: ZodError<Input>;
+    data?: undefined // Modified this to make it easier to access the data
 };
 export declare type SafeParseReturnType<Input, Output> = SafeParseSuccess<Output> | SafeParseError<Input>;
 export declare abstract class ZodType<Output = any, Def extends ZodTypeDef = ZodTypeDef, Input = Output> {
